tests/test_smoke.py

```diff
@@ -2625,13 +2626,63 @@ def test_user_ray_cluster(generic_cloud: str):
     run_one_test(test)
 
 
+_CODE_PREFIX = ['import sky']
+
+
+def _build(code: List[str]) -> str:
+    code = _CODE_PREFIX + code
+    code = ';'.join(code)
+    return f'python3 -u -c {shlex.quote(code)}'
+
+
 # ------- Testing the core API --------
 # Most of the core APIs have been tested in the CLI tests.
 # These tests are for testing the return value of the APIs not fully used in CLI.
-def test_core_api():
-    name = _get_cluster_name()
-    sky.launch
-    # TODO(zhwu): Add a test for core api.
+
+
+def test_core_api_sky_launch_dryrun():
+    code = [
+        'task = sky.Task()',
+        'job_id, handle = sky.launch(task, dryrun=True)',
+        'assert job_id is None and handle is None',
+    ]
+    cmd = _build(code)
+    test = Test(
+        'core-api-sky-launch-dryrun',
+        [cmd],
+    )
+    run_one_test(test)
```

Comment:
As we are doing a dryrun, it would be nice to include it in the unit pytests with a programmetic way instead of smoke test.